### PR TITLE
Implement Alert Feature for Unsaved Changes in Editor

### LINKED ISSUE #11: Better save status
Actually show if there's been unsaved changes in the editor view and warn the user if they try to leave the page with unsaved changes.

### PR DESCRIPTION
**Pull request by mSaurabh**
This pull request implements the "Better Save" feature (issue #11) by introducing an alert mechanism that warns users of unsaved changes when attempting to leave the screen without saving their content. The changes are contained within the editor.js file.

**Changes**:
Added a new alert feature that triggers when the user attempts to navigate away from the editor screen without saving their changes.
Implemented a check to detect unsaved changes and display a confirmation prompt to the user.

**Related Issues:**
Closes #11: Better Save feature implementation

**Notes:**
This implementation uses a flag called `prevent_save` and function `checkForUnsavedChanges` to check for unsaved changes. Also added a `beforeunload` event listener to make changes to prevent_save flag based on user choice.

### CODE REVIEW

File: cardie/static/main/scripts/editor/editor.js [hunk 1]
```diff
@@ -37,16 +41,19 @@ async function start_editor() {
 
                 } else {
                     status_error();
-                    log("WARNING", "There was a problem")
-                    create_notification("There was a problem", "There was an unknown issue", "warning");
+                    log("WARNING", "There was a problem");
```
Comment (mSaurabh) [inline cardie/static/main/scripts/editor/editor.js:44]: Formatter being naughty. 😅

**Comment by nfoert** [comment]
Hey @mSaurabh, thanks for working on this! I attempted to test out the new feature but it appears that most of the static files couldn't be loaded anymore
image
```
The resource from “http://127.0.0.1:8000/static/main/ui.css” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff). 127.0.0.1:8000
The resource from “http://127.0.0.1:8000/static/main/footer.css” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff). 127.0.0.1:8000
The resource from “http://127.0.0.1:8000/static/main/scripts/global/favicon.js” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff). 127.0.0.1:8000
--snip--
```
Is this happening for you too or is this just some kind of error on my side? I'm not sure why this is occurring as you've not committed to anything besides `editor.js`

**Comment by mSaurabh** [comment]
I think ar4s added some ruff changes that I pulled from master branch before creating my PR that might be causing these errors

**Comment by nfoert** [comment]
> I think ar4s added some ruff changes that I pulled from master branch before creating my PR that might be causing these errors

It turns out I had forgotten to create the `.env` file which was causing some strange behaviors, that was my bad

I also added a simple indicator to show if there's been any unsaved changes as I think that would be a good visual cue to the user
image

I'll get this merged now, thanks again!

**Pull request merged**